e2e/images/Dockerfile

```diff
@@ -0,0 +1,83 @@
+FROM ubuntu:16.04
+ENV container docker
+ENV LC_ALL C
+ENV DEBIAN_FRONTEND noninteractive
+
+#RUN sed -i "s/^exit 101$/exit 0/" /usr/sbin/policy-rc.d
+
+# Install common dependencies
+RUN apt-get -y clean && apt-get -y update && apt-get install -qqyf \
+    ca-certificates \
+    lxc \
+    iptables \
+    jq \
+    software-properties-common \
+    apt-transport-https \
+    curl \
+    git
+
+
+#Delete systemd services we do not need
+RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
+    rm -f /lib/systemd/system/multi-user.target.wants/*;\
+    rm -f /etc/systemd/system/*.wants/*;\
+    rm -f /lib/systemd/system/local-fs.target.wants/*; \
+    rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
+    rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
+    rm -f /lib/systemd/system/basic.target.wants/*;\
+    rm -f /lib/systemd/system/anaconda.target.wants/*;
+
+
+RUN apt-get update &&\
+    apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D &&\
+    apt-add-repository 'deb https://apt.dockerproject.org/repo ubuntu-xenial main' &&\
```

Comment:
This is how kubeadm installs docker 1.12. The add a source file from kubernetes. Perhaps we should do the same since E2E uses kubeadm to setup K8S.
https://github.com/luxas/kubeadm-workshop#installing-kubeadm-on-all-the-machines-you-want-in-your-cluster.